File: TheLionsDen.Model/Responses/UserResponse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TheLionsDen.Model.Responses
{
    public class UserResponse
    {
        public int UserId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Username { get; set; } 
        public string Email { get; set; } 
        public string PhoneNumber { get; set; }
        public string Status { get; set; } 
        public DateTime? DateOfBirth { get; set; }
        public string Gender { get; set; }
        public int RoleId { get; set; }

        public virtual RoleResponse Role { get; set; }
        //public virtual ICollection<Favourite> Favourites { get; set; }
        //public virtual ICollection<Reservation> Reservations { get; set; }

        public string FullName => $"{FirstName} {LastName}";
        public string RoleName => Role.Name;
    }
}

File: the_lions_den_mobile/lib/model/user/user_response.dart
import 'package:json_annotation/json_annotation.dart';
part 'user_response.g.dart';

@JsonSerializable()
class UserResponse {
  int? userId;
  String? firstName;
  String? lastName;
  String? username;
  String? email;
  String? phoneNumber;
  String? status;
  DateTime? dateOfBirth;
  String? gender;
  int? roleId;

  String? fullName;
  String? roleName;

  UserResponse();

  factory UserResponse.fromJson(Map<String, dynamic> json) =>
      _$UserResponseFromJson(json);

  /// Connect the generated [_$UserResponseToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$UserResponseToJson(this);
}
